### PR TITLE
feat: Model/schema additions for admin billing info

### LINKED ISSUE #6097: Add details for admin invoice billing
**Describe the feature**
Related front-end PR:https://github.com/fossasia/open-event-frontend/pull/3152
We need model additions for admin invoice billing info in settings so that the organisers can pay their invoice to the admin.

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6097 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.
#### Changes proposed in this pull request:

Adds admin invoice details
* schema additions
* model additions
* migration file

### CODE REVIEW

File: migrations/versions/e1e80bcf4343_.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+# revision identifiers, used by Alembic.
+revision = 'e1e80bcf4343'
+down_revision = '43e8c59337ae'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('settings', sa.Column('admin_additional_info', sa.String(), nullable=True))
```
Comment (mrsaicharan1) [inline migrations/versions/e1e80bcf4343_.py]: @shreyanshdwivedi @uds5501 @iamareebjamal These fields don't require a default value right? Please correct me if I'm wrong.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 Please review.

**Comment by mrsaicharan1** [comment]
Travis has passed for this but It still shows it's in progress.

File: migrations/versions/e1e80bcf4343_.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    op.add_column('settings', sa.Column('admin_company', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_company_address', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_company_city', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_company_comments', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_contact_name', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_country', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_email', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_phone', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_tax_info', sa.String(), server_default='', nullable=True))
+    op.add_column('settings', sa.Column('admin_zip', sa.String(), server_default='', nullable=True))
```
Comment (iamareebjamal) [inline migrations/versions/e1e80bcf4343_.py]: No need for server default. NULL is better

**Comment by mrsaicharan1** [comment]
@iamareebjamal Removed the server defaults. Please review.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 @iamareebjamal Removed the server defaults. Please have a look.

File: app/models/setting.py [hunk 3]
```diff
@@ -197,7 +211,18 @@ def __init__(self,
                  omise_live_public=None,
                  omise_live_secret=None,
                  alipay_publishable_key=None,
-                 alipay_secret_key=None):
```
Comment (mrsaicharan1) [inline app/models/setting.py]: This should be here. Will restore this.

File: app/api/schema/settings.py [hunk 4]
```diff
@@ -196,3 +196,16 @@ class Meta:
...
     smtp_port = fields.Integer(allow_none=True)
     smtp_encryption = fields.Str(allow_none=True)  # Can be tls, ssl, none
+
+    # Admin Invoice Details
+    admin_contact_name = fields.Str(allow_none=True)
+    admin_phone = fields.Str(allow_none=True)
+    admin_email = fields.Str(allow_none=True)
+    admin_country = fields.Str(allow_none=True)
+    admin_tax_info = fields.Str(allow_none=True)
+    admin_company = fields.Str(allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/settings.py]: instead of _company, let's follow the billing naming convention, it's more accurate.

Comment (mrsaicharan1) [reply app/api/schema/settings.py]: In this context, do you mean `admin_billing_company` ?

File: app/api/schema/settings.py [hunk 5]
```diff
@@ -196,3 +196,16 @@ class Meta:
     smtp_password = fields.Str(allow_none=True)
     smtp_port = fields.Integer(allow_none=True)
     smtp_encryption = fields.Str(allow_none=True)  # Can be tls, ssl, none
+
+    # Admin Invoice Details
+    admin_contact_name = fields.Str(allow_none=True)
+    admin_phone = fields.Str(allow_none=True)
+    admin_email = fields.Str(allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/settings.py]: this field is not required.

File: app/api/schema/settings.py (hunk 4 above)
Comment (CosmicCoder96) [reply app/api/schema/settings.py]: um, rest of the fields, not this one. so let is be admin_compnay,

make others admin_billing_xxxx like we are doing for the other PR.

**Comment by mrsaicharan1** [comment]
@iamareebjamal @CosmicCoder96 @uds5501 @shreyanshdwivedi Please review.

File: app/api/schema/settings.py (hunk 4 above)
Comment (mrsaicharan1) [reply app/api/schema/settings.py]: Outdated

File: app/api/schema/settings.py [hunk 6]
```diff
@@ -196,3 +196,16 @@ class Meta:
...
     smtp_encryption = fields.Str(allow_none=True)  # Can be tls, ssl, none
+
+    # Admin Invoice Details
+    admin_billing_contact_name = fields.Str(allow_none=True)
+    admin_billing_phone = fields.Str(allow_none=True)
+    admin_billing_email = fields.Email(allow_none=True)
+    admin_billing_country = fields.Str(allow_none=True)
+    admin_billing_tax_info = fields.Str(allow_none=True)
+    admin_company = fields.Str(allow_none=True)
+    admin_billing_company_address = fields.Str(allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/settings.py]: admin_billing_address

File: app/api/schema/settings.py [hunk 7]
```diff
@@ -196,3 +196,16 @@ class Meta:
...
+    admin_billing_company_city = fields.Str(allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/settings.py]: admin_billing_city

File: app/api/schema/settings.py [hunk 8]
```diff
@@ -196,3 +196,16 @@ class Meta:
...
+    admin_billing_zip = fields.Str(allow_none=True)
+    admin_billing_additional_info = fields.Str(allow_none=True)
+    admin_billing_company_comments = fields.Str(allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/settings.py]: admin_billing_additional_info

Comment (mrsaicharan1) [reply app/api/schema/settings.py]: Already done!

File: app/api/schema/settings.py (hunk 7 above)
Comment (mrsaicharan1) [reply app/api/schema/settings.py]: Outdated

File: app/api/schema/settings.py (hunk 8 above)
Comment (CosmicCoder96) [reply app/api/schema/settings.py]: @mrsaicharan1 https://github.com/fossasia/open-event-server/blob/85f7fa80f8fcc229ed16a7d8d9feaf40f6d34edb/app/api/schema/settings.py

it's not reflecting.

File: app/api/schema/settings.py (hunk 7 above)
Comment (iamareebjamal) [reply app/api/schema/settings.py]: @mrsaicharan1 It'll automatically show outdated once you have done the change

File: app/api/schema/settings.py (hunk 8 above)
Comment (iamareebjamal) [reply app/api/schema/settings.py]: @mrsaicharan1 It'll automatically show outdated once you have done the change

**Comment by mrsaicharan1** [comment]
Okay, might be something on my side. Will check

File: app/api/schema/settings.py (hunk 8 above)
Comment (mrsaicharan1) [reply app/api/schema/settings.py]: @CosmicCoder96 This is the required change?` admin_billing_additional_info`

Comment (mrsaicharan1) [reply app/api/schema/settings.py]: https://github.com/fossasia/open-event-server/blob/9ed109f0f2849df0968a07adf98c3e483210a755/app/api/schema/settings.py
@CosmicCoder96 Please have a look here

**Comment by mrsaicharan1** [comment]
https://github.com/fossasia/open-event-server/blob/9ed109f0f2849df0968a07adf98c3e483210a755/app/api/schema/settings.py
@CosmicCoder96 Please have a look here

File: app/api/schema/settings.py (hunk 8 above)
Comment (CosmicCoder96) [reply app/api/schema/settings.py]: @mrsaicharan1 . https://github.com/fossasia/open-event-server/blob/9ed109f0f2849df0968a07adf98c3e483210a755/app/api/schema/settings.py#L211

You are repearting this field. `admin_billing_comapany_comments`  has to be renamed to `admin_billing_additional_info` a new field is not required. That was the originally requested change.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Please review.

File: app/models/setting.py [hunk 9]
```diff
@@ -197,7 +210,17 @@ def __init__(self,
...
+                 alipay_secret_key=None,
+                 admin_billing_contact_name=None,
+                 admin_billing_phone=None,
+                 admin_billing_email=None,
+                 admin_billing_country=None,
+                 admin_billing_tax_info=None,
+                 admin_company=None,
+                 admin_billing_address=None,
+                 admin_billing_city=None,
+                 admin_billing_company_zip=None,
```
Comment (CosmicCoder96) [inline app/models/setting.py]: This is not correct.

File: app/models/setting.py [hunk 10]
```diff
@@ -268,6 +291,17 @@ def __init__(self,
...
 
+        # Admin Invoice Details
+        self.admin_billing_contact_name = admin_billing_contact_name
+        self.admin_billing_phone = admin_billing_phone
+        self.admin_billing_country = admin_billing_country
+        self.admin_billing_tax_info = admin_billing_tax_info
+        self.admin_company = admin_company
+        self.admin_billing_address = admin_billing_address
+        self.admin_billing_city = admin_billing_city
+        self.admin_billing_company_zip = admin_billing_company_zip
```
Comment (CosmicCoder96) [inline app/models/setting.py]: This is not correct.

File: app/models/setting.py (hunk 9 above)
Comment (CosmicCoder96) [reply app/models/setting.py]: it needs to match the definition inside schema/model.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 Please do a peer review.

**Comment by iamareebjamal** [comment]
There's too much repetition. Open an issue to use dataclasses in future: 
https://realpython.com/python-data-classes/

**Pull request merged**